Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/MotionBlur.azsl

```diff
@@ -0,0 +1,176 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <scenesrg_all.srgi>
+#include <Atom/Features/PostProcessing/PostProcessUtil.azsli>
+#include "MotionBlur.azsli"
+
+#define TILE_DIM_X 16
+#define TILE_DIM_Y 16
+
+ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
+{
+    Texture2D<float4> m_inputColor;
+    Texture2D<float4> m_linearDepth;
+    Texture2D<float2> m_inputVelocity;
+    Texture2D<float2> m_inputTileVelocity;
+
+    RWTexture2D<float4> m_outputColor;
+
+    struct Constants
+    {
+        uint2 m_outputSize; // Texture size of output
+        float m_strength; // Multiplier for sampling offset
+        [[pad_to(16)]]
+    };
+    Constants m_constants;
+}
+
+option enum class SampleNumber
+{
+    Sample6,
+    Sample14,
+    Sample24,
+    Sample36
+}
+o_sampleNumber = SampleNumber::Sample14;
+
+float2 DepthCmp(float centerDepth, float sampleDepth, float depthScale)
+{
+    return saturate(0.5f + float2(depthScale, -depthScale) * (sampleDepth - centerDepth));
+}
+
+float2 SpreadCmp(float offsetLen, float2 spreadLen, float pixelToSampleUnitsScale)
+{
+    return saturate(pixelToSampleUnitsScale * spreadLen - offsetLen + 1.0f);
+}
+
+float SampleWeight(float centerDepth, float sampleDepth, float offsetLen, float centerSpreadLen, float sampleSpreadLen, float pixelToSampleUnitsScale, float depthScale)
+{
+    float2 depthCmp = DepthCmp(centerDepth, sampleDepth, depthScale);
+    float2 spreadCmp = SpreadCmp(offsetLen, float2(centerSpreadLen, sampleSpreadLen), pixelToSampleUnitsScale);
+    return dot(depthCmp, spreadCmp);
+}
+
+
+uint2 ClampCoords(int2 coord)
+{
+    uint2 clampedCoords;
+    clampedCoords.x = uint(clamp(coord.x, 0, PassSrg::m_constants.m_outputSize.x - 1));
+    clampedCoords.y = uint(clamp(coord.y, 0, PassSrg::m_constants.m_outputSize.y - 1));
+    return clampedCoords;
+}
+
+float2 ToPixelVelocity(float2 velocity)
+{
+    velocity *= float2(PassSrg::m_constants.m_outputSize);
+    velocity *= PassSrg::m_constants.m_strength;
+    return velocity;
+}
+
+// This shader calculates motion blur effect from tile max speeds. 
+// See https://advances.realtimerendering.com/s2014/index.html#_NEXT_GENERATION_POST
+// for a detailed overview of the technique. 
+[numthreads(TILE_DIM_X, TILE_DIM_Y, 1)]
+void MainCS(
+    uint3 dispatchThreadID : SV_DispatchThreadID, 
+    uint3 groupID : SV_GroupID, 
+    uint groupIndex : SV_GroupIndex)
+{
+    uint2 pixelCoord = dispatchThreadID.xy;
+    if (pixelCoord.x >= PassSrg::m_constants.m_outputSize.x || pixelCoord.y >= PassSrg::m_constants.m_outputSize.y)
+    {
+        return;
+    }
+
+    float3 color = PassSrg::m_inputColor[pixelCoord].rgb;
+    PassSrg::m_outputColor[pixelCoord].rgb = color;
+
+    float depth = PassSrg::m_linearDepth[pixelCoord].r;
+
+    float2 tileVelocity = PassSrg::m_inputTileVelocity[pixelCoord / uint2(TILE_SIZE, TILE_SIZE)];
+    tileVelocity = ToPixelVelocity(tileVelocity);
```

Comment:
I think you need to add some noise to tileVelocity. I think the slides mention it too as well as this shader - https://github.com/ValtoGameEngines/CryEngine/blob/d9d2c9f000836f0676e65a90bed40dcc3b1451eb/Engine/Shaders/HWScripts/CryFX/MotionBlur.cfx#L73. Are we able to do the same?

Reply:
Oh I missed that, thanks for reminding me.